routes/web.php

```diff
@@ -21,9 +21,14 @@
 $router->get('campaigns/{id}/inbox', 'Legacy\Web\AshesCampaignsController@showInbox');
 $router->get('campaigns/{id}', 'Legacy\Web\AshesCampaignsController@showCampaign')->name('campaigns.show');
 // Create, update, delete campaigns via Rogue.
-$router->post('campaigns', 'Legacy\Web\CampaignsController@store');
-$router->patch('campaigns/{campaign}', 'Legacy\Web\CampaignsController@update');
-$router->delete('campaigns/{campaign}', 'Legacy\Web\CampaignsController@destroy');
+// @TODO: rename these routes (to just /campaigns) when we get off Ashes.
+$router->get('campaign-ids/create', ['as' => 'campaign_id.create', 'uses' => 'Legacy\Web\CampaignsController@create']);
```

Comment:
These can all be consolidated into a resource route:

```php
$router->resource('campaign-ids', 'Legacy\Web\CampaignsController');
```

Reply:
I believe because `campaign-id` isn't the actual model name, you'll need to [explicitly bind it on the model](https://laravel.com/docs/5.7/routing#route-model-binding):

```php
/**
 * Get the route key for the model.
 *
 * @return string
 */
public function getRouteKeyName()
{
    return 'campaign-ids';
}
```